### DISCUSSION TITLE
Idea for v2, v3, and afterwards

### DISCUSSION THREAD

**Discussion by jannikbuschke** [Hellenic/react-hexgrid#88] [Ideas]
Maybe we can gather here some ideas for future developments.

My ideas and rough roadmap could be

V2 (mostly done):
 - add typescript
 - refactor to use newer react context
 - add samples in storybook
 - basically no breaking changes

V3
- improve svg primitives and low level api (probably with breaking changes)
- add svg documentation/knowledge

V4/ongoing
- have a clear primitive core package with good primitives and additional feature packages
- css/styling
- drag and drop

**Comment by tieje** [comment]
I totally agree about improving svg primitives and low level api, but I don't really know what improvements you're talking about. I'll look into it today.

**Reply by jannikbuschke** [reply]
Maybe the low level api is also pretty ok as it is but when I started using this lib it was necessary to look at the source code to do some styling. The codebase is nice/small/good consumable, however it is necessary to actually checkout the source code and investiage how the svg elements are layed out in the dom hierarchy. I think the api can be improved, so that just by looking at the code it is clear what svg elements are used, how to style theme and what ui transformations this lib applies.
Also I think drag and drop and the pinch/zooming you mentioned in discord could be factored in.
